tests/components/command_line/test_binary_sensor.py

```diff
@@ -101,7 +101,7 @@ async def test_template(hass: HomeAssistant, load_yaml_integration: None) -> Non
     entity_state = hass.states.get("binary_sensor.test")
     assert entity_state
     assert entity_state.state == STATE_ON
-    assert entity_state.attributes.get("icon") == "mdi:on"
+    assert entity_state.attributes.get("icon") == "mdi:airplane-takeoff"
```

Comment:
I dont understand this. The test had a template checking if its state was on , the created a icon of on.

If that doesnt work anymore this ends up a rather ugly breaking change?

Reply:
No, as the behavior was incorrect before so it's a bugfix.
`this` refers to the current state, not the state that is currently being evaluated.

I was very confused about this myself hence I'm now fixing it here + tried to make it clearer in the docs (https://www.home-assistant.io/integrations/template/#template-and-action-variables)

Reply:
Ok. So the tests should probably be rewritten then. To make sense in the new context. They felt complicated now.

Also i still think this ends up being a breaking change, even if it was not intended behavior. We even had tests to verify the incorrect behavior.

Might need some more eyes on this.

Reply:
Updated the examples again to be more generic and not so attached to the state.

I have a hard time seeing this as breaking as it's about templating and the use of `this`. `ManualTriggerEntity` is only a minor part in this so while yes, it might break someones workflow but I think it would be very minor (as most use of `this` I don't think is part of this).